src/clj_gatling/simulation.clj

```diff
@@ -92,27 +92,37 @@
         step-ctx [(:steps scenario) (:step-fn scenario)]]
     (go-loop [[step context step-ctx] (next-step step-ctx final-context)
               results []]
-             (let [[result new-ctx] (<! (async-function-with-timeout step
-                                                                     timeout
-                                                                     sent-requests
-                                                                     user-id
-                                                                     context))
-                   [step' _ _ :as next-steps] (next-step step-ctx new-ctx)]
-               (when-let [e (:exception result)]
-                 (log-exception (:error-file options) e))
-               (if (or (should-terminate?)
-                       (nil? step')
-                       (and skip-next-after-failure?
-                            (request-failed? result)))
-                 (do
-                   (when post-hook
-                     (post-hook context))
-                   (>! result-channel (->> (dissoc result :exception)
-                                           (conj results))))
-                 (recur next-steps (conj results result)))))
+      (let [[result new-ctx] (<! (async-function-with-timeout step
+                                                              timeout
+                                                              sent-requests
+                                                              user-id
+                                                              context))
+            [step' _ _ :as next-steps] (next-step step-ctx new-ctx)]
+        (when-let [e (:exception result)]
+          (log-exception (:error-file options) e))
+        (if (or (should-terminate?)
+                (nil? step')
+                (and skip-next-after-failure?
+                     (request-failed? result)))
+          (do
+            (when post-hook
+              (post-hook context))
+            (>! result-channel (->> (dissoc result :exception)
+                                    (conj results))))
+          (recur next-steps (conj results result)))))
     result-channel))
 
-(defn- run-scenario-constantly
+(defn arg-count [f]
```

Comment:
Could you move this utility function to `simulation-util` namespace? `simulation` namespace is already quite complex and I think it helps if all general utility functions are defined elsewhere.

Reply:
Absolutely! Do you think `rate->interval-ns` and `run-at` should be moved too?

Reply:
Done 👍